tenant-cd/src/main/java/ai/vespa/hosted/cd/TestConfig.java

```diff
@@ -0,0 +1,101 @@
+package ai.vespa.hosted.cd;
+
+import ai.vespa.hosted.api.ControllerHttpClient;
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.SystemName;
+import com.yahoo.slime.ArrayTraverser;
+import com.yahoo.slime.Inspector;
+import com.yahoo.slime.JsonDecoder;
+import com.yahoo.slime.ObjectTraverser;
+import com.yahoo.config.provision.zone.ZoneId;
+import com.yahoo.slime.Slime;
+
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The place to obtain environment-dependent configuration for the current test run.
+ *
+ * If the system property 'vespa.test.config' is set, this class attempts to parse config
+ * from a JSON file at that location -- otherwise, attempts to access the config will return null.
+ *
+ * @author jvenstad
+ */
+public class TestConfig {
+
+    private static TestConfig theConfig;
+
+    private final ApplicationId application;
+    private final ZoneId zone;
+    private final SystemName system;
+    private final Map<ZoneId, Deployment> deployments;
+
+    private TestConfig(ApplicationId application, ZoneId zone, SystemName system, Map<ZoneId, Deployment> deployments) {
+        this.application = application;
+        this.zone = zone;
+        this.system = system;
+        this.deployments = Map.copyOf(deployments);
+    }
+
+    /** Returns the config for this test, or null if it has not been provided. */
+    public static synchronized TestConfig get() {
+        if (theConfig == null) {
+            String configPath = System.getProperty("vespa.test.config");
+            theConfig = configPath != null ? fromFile(configPath) : fromController();
+        }
+        return theConfig;
+    }
+
+    /** Returns the full id of the application to be tested. */
+    public ApplicationId application() { return application; }
+
+    /** Returns the zone of the deployment to test. */
+    public ZoneId zone() { return zone; }
```

Comment:
Consider returning something other than ZoneId, and defined in this package, to allow adding more properties than environment and region later. (As a side-node, ZoneId will hopefully change to become just env/region in the near future)